src/test/java/unit/AiNotebookControlled.java

```diff
@@ -0,0 +1,312 @@
+  @Test
+  @DisplayName("list and describe reflect creating and deleting a controlled notebook")
+  void listDescribeReflectCreateDelete() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra resources create ai-notebook --name=$name`
+    String name = "listDescribeReflectCreateDelete";
+    UFAiNotebook createdNotebook =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFAiNotebook.class, "resources", "create", "ai-notebook", "--name=" + name);
+
+    // check that the name and notebook name match
+    assertEquals(name, createdNotebook.name, "create output matches name");
+
+    // ai notebooks are always private
+    assertEquals(
+        AccessScope.PRIVATE_ACCESS, createdNotebook.accessScope, "create output matches access");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        createdNotebook.privateUserName.toLowerCase(),
+        "create output matches private user name");
+    // TODO (PF-616): check the private user roles once WSM returns them
+
+    // check that the notebook is in the list
+    UFAiNotebook matchedResource = listOneNotebookResourceWithName(name);
+    assertEquals(name, matchedResource.name, "list output matches name");
+
+    // `terra resources describe --name=$name --format=json`
+    UFAiNotebook describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFAiNotebook.class, "resources", "describe", "--name=" + name);
+
+    // check that the name matches and the instance id is populated
+    assertEquals(name, describeResource.name, "describe resource output matches name");
+    assertNotNull(describeResource.instanceName, "describe resource output includes instance name");
+
+    // ai notebooks are always private
+    assertEquals(
+        AccessScope.PRIVATE_ACCESS, describeResource.accessScope, "describe output matches access");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        describeResource.privateUserName.toLowerCase(),
+        "describe output matches private user name");
+    // TODO (PF-616): check the private user roles once WSM returns them
+
+    // `terra notebooks delete --name=$name`
+    TestCommand.Result cmd = TestCommand.runCommand("resources", "delete", "--name=" + name);
+    assertThat(
+        "delete either succeeds or times out",
+        cmd.exitCode == 0
+            || (cmd.exitCode == 1
+                && cmd.stdErr.contains(
+                    "CLI timed out waiting for the job to complete. It's still running on the server.")));
+
+    // confirm it no longer appears in the resources list
+    List<UFAiNotebook> listedNotebooks = listNotebookResourcesWithName(name);
+    assertEquals(
+        0, listedNotebooks.size(), "deleted notebook no longer appears in the resources list");
```

Comment:
nit: `assertThat( "...", listedNotebooks, Matchers.empty());` is a little more idiomatic and will show the contents of listedNotebooks if it's not empty.

Reply:
Good tip! I didn't know about that one.